operator/pipelineutil/exoscale_connector.go

```diff
@@ -0,0 +1,99 @@
+package pipelineutil
+
+import (
+	"context"
+	"fmt"
+
+	pipeline "github.com/ccremer/go-command-pipeline"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	exoscalesdk "github.com/exoscale/egoscale/v2"
+	providerv1 "github.com/vshn/provider-exoscale/apis/provider/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ExoscaleAPIKey identifies the key in which the API Key of the exoscale.com is expected in a Secret.
+	ExoscaleAPIKey = "EXOSCALE_API_KEY"
+	// ExoscaleAPISecret identifies the secret in which the API Secret of the exoscale.com is expected in a Secret.
+	ExoscaleAPISecret = "EXOSCALE_API_SECRET"
+)
+
+type ExoscaleConnector struct {
+	ApiKey    string
+	ApiSecret string
+	Exoscale  *exoscalesdk.Client
+}
+
+type connectContext struct {
+	context.Context
+	kube               client.Client
+	ProviderConfigName string
+
+	providerConfig   *providerv1.ProviderConfig
+	credentialSecret *corev1.Secret
+	apiKey           string
+	apiSecret        string
+	exoscaleClient   *exoscalesdk.Client
+	opts             []exoscalesdk.ClientOpt
+}
+
+// OpenExoscaleClient fetches the ProviderConfig given by the name, fetches the API secret and returns a ExoscaleConnector with an initialized client.
+func OpenExoscaleClient(ctx context.Context, kube client.Client, providerConfigRef string, opts ...exoscalesdk.ClientOpt) (*ExoscaleConnector, error) {
```

Comment:
This should probably go in the follow-up PR?

Reply:
I added a commit that refactors some code. but I _may_ be able to reorder the commits